pkg/util/configstore.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2018 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"errors"
+	"fmt"
+	"k8s.io/klog"
+	"path"
+	"strings"
+)
+
+// StoreReader interface enables plugging different stores, that contain the
+// keys and data. (e.g k8s secrets or local files)
+type StoreReader interface {
+	DataForKey(clusterID string, key string) (string, error)
+}
+
+/* ConfigKeys contents and format,
+- csMonitors: MON list, comma separated
+- csAdminID: adminID, used for provisioning
+- csUserID: userID, used for publishing
+- csAdminKey: key, for userID in csProvisionerUser
+- csUserKey: key, for userID in csPublisherUser
+- csPools: Pool list, comma separated
+*/
+
+// Constants for various ConfigKeys
+const (
+	csMonitors = "monitors"
+	csAdminID  = "adminid"
+	csUserID   = "userid"
+	csAdminKey = "adminkey"
+	csUserKey  = "userkey"
+	csPools    = "pools"
+)
+
+// ConfigStore provides various gettors for ConfigKeys
+type ConfigStore struct {
+	StoreReader
+}
+
+// dataForKey returns data from the config store for the provided key
+func (dc *ConfigStore) dataForKey(clusterID string, key string) (string, error) {
```

Comment:
`(dc *ConfigStore) dataForKey(clusterID string, key string) (string, error)`
 to
 `(dc *ConfigStore) dataForKey(clusterID , key string) (string, error)`